### PR TITLE
Move to Spring for Authentication.

### PR DESCRIPTION
**Pull request by BarDweller**
This is quite the change.. moves us off the Liberty + Custom Libs auth impl to using Spring Security's OAuth2 support.

Note this PR requires gameontext/gameon#152
The PR 152 enables the 2nd domain used by the new dummy auth module within this PR.
This does not affect production, as production does not deploy the dummy auth module.

Also, this still needs testing with Docker/Docker-Compose and non OpenShift K8S's.

Most notably the dummy auth provider is now a full blown local impl of an OAuth2 Provider, (albeit a fairly dim one that's not going to win any prizes). That brought with it some requirements re how that is accessed that were unexpected. The host name for that provider MUST be different to the host name for the application, because OAuth2 uses Cookies, and Cookies work with host names, and if they aren't different.. it won't work. Adding a new host for OpenShift means adding a new route to accomodate the new host. New hosts for docker/minikube will likely mean we need to add a different host.ipaddress.xip.io to our properties somewhere.. that'll be for the root project script updates to handle.

Need to test via Docker/Minikube before this becomes non-draft

---
This change is [Reviewable](https://reviewable.io/reviews/gameontext/gameon-auth/41)

### CODE REVIEW

**Comment by BarDweller** [comment]
Docker tested OK
Minikube/Helm/Istio tested OK

File: README.md [hunk 1]
```diff
@@ -1,18 +1,44 @@
...
+The appropriate information is then retrieved from the Spring Security authentication, and is used to build the JWT to return to the user.
+
+*Note:* All urls need to start /auth to emulate the Old-Auth context root approach, otherwise GameOn Proxy would need updating to know how to route traffic to this service. As this is currently intended to be a drop in replacement, to enable A/B testing, canary deployment etc, it was better to keep urls compatible with Old-Auth. This also includes acutators which are moved to `/auth` in this project, eg `/auth/health` )
+
+*Note:* Had to add a secondary domain name used to access the `/auth/dummy` endpoint because that's emulating an entirely different oauth2 server, we may yet revisit this 
+idea, but that's how it sits for now. 
+
+Twitter isn't an OAuth2 provider, so the entire Twitter flow is handled by a TwitterController (RestController) that uses twitter4j to do the login, based on code from the old auth impl.
 
-## Contributing
```
Comment (ebullient) [inline README.md:14]: Keep this section, please

File: README.md [hunk 2]
```diff
@@ -1,18 +1,44 @@
-# Game On! Auth Service 
+# Spring Security GameOn Auth.
 
-[![Codacy Badge](https://api.codacy.com/project/badge/Grade/bfd8ece69df0405eb5165f00fefc087e)](https://www.codacy.com/app/gameontext/gameon-auth?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=gameontext/gameon-auth&amp;utm_campaign=Badge_Grade)
```
Comment (ebullient) [inline README.md:3]: Is codacy broken?

File: README.md [hunk 3]
```diff
@@ -1,18 +1,44 @@
...
+Implementation of GameOn Auth project, using Spring Security.
 
-See the [application architecture description](https://gameontext.gitbooks.io/gameon-gitbook/content/microservices/) in the Game On! Docs for more information on how to use this service.
```
Comment (ebullient) [inline README.md]: This section in the book should perhaps be updated, and we could change this line to say "for more information about how this service fits in the overall application architecture."

File: auth-app/build.gradle [hunk 4]
```diff
@@ -1,101 +1,36 @@
...
-  toolVersion = "0.8.4"
+plugins {
+	id 'org.springframework.boot' version '2.3.2.RELEASE'
+	id 'io.spring.dependency-management' version '1.0.9.RELEASE'
+	id 'java'
 }
 
-sourceCompatibility = 1.8
+group = 'gameontext'
+version = '0.0.1-SNAPSHOT'
```
Comment (ebullient) [inline auth-app/build.gradle:14]: we're at 0.0.7 at least.. :-P
In seriousness, we should probably version all of these apps beyond 0.0.1 ..

File: auth-app/build.gradle [hunk 5]
```diff
@@ -1,101 +1,36 @@
...
+	
+	testImplementation('org.springframework.boot:spring-boot-starter-test') {
+		exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
+	}
+	testImplementation 'org.springframework.kafka:spring-kafka-test'
+	testImplementation 'org.springframework.security:spring-security-test'
 }
 
 test {
-   jvmArgs "-javaagent:${classpath.find { it.name.contains("jmockit") }.absolutePath}"
```
Comment (ebullient) [inline auth-app/build.gradle:70]: AH! you removed jacoco test coverage.. that's why you dropped codacy. Could you please keep it? We should be testing w/ coverage.

File: README.md (hunk 1 above)
Comment (BarDweller) [reply README.md:14]: Done.

File: auth-app/build.gradle (hunk 4 above)
Comment (BarDweller) [reply auth-app/build.gradle:14]: Aye maybe, not this time round tho.

File: auth-app/build.gradle (hunk 5 above)
Comment (BarDweller) [reply auth-app/build.gradle:70]: Added back

File: README.md (hunk 3 above)
Comment (BarDweller) [reply README.md]: Sounds like something that will need addressing across every GO service, they all share this boilerplate.. separate issue to the spring move.

File: README.md (hunk 2 above)
Comment (BarDweller) [reply README.md:3]: Codacy went on a short vacation when I used the auth-spring-social project as the template for this one, which meant the gradle & readme lost jacoco/codacy. Restored now =)

**Review by ebullient — APPROVED** [review]

**Pull request merged**